### DISCUSSION TITLE
Do PnP Site Templates support copying files in document libraries?

### DISCUSSION THREAD

**Discussion by carlgrzywacz** [pnp/powershell#919] [Q&A] [answered]
I created an out of the box document library, added a few files and ran the PowerShell below.  The unzipped schema.pnp file does not contain any of the files in the document library.  I also receive an error when running the Invoke-PnpSiteTemplate cmdlet.

Error: Invoke-PnPSiteTemplate : Input string was not in a correct format.

Get-PnPSiteTemplate -Out $listSchmaXMLPath -Handlers Lists -ListsToExtract "TestLibrary" 
Add-PnPDataRowsToSiteTemplate -Path $listSchmaXMLPath -List 'TestLibrary' 
Connect-PnPOnline -Url $destinationSiteURL -Interactive
Invoke-PnPSiteTemplate -Path $listSchmaXMLPath -Handlers Lists

Is this a bug or an unsupported action?

**Comment by kasperbolarsen** [comment, marked as answer]
Add-PnPDataRowsToSiteTemplate is, as far as I know, intented to add listitems to the template, not files. This might also explain why you get the error when applying the template to the site.
Add-PnPFileToProvisioningTemplate might do the trick :-)

**Reply by MrM40** [reply]
Please correct spelling from `Add-PnPDateRowsToSiteTemplate` to `Add-PnPDataRowsToSiteTemplate` (Date > Data) ;-)
And I assume `Add-PnPFileToProvisioningTemplate` is now `Add-PnPFileToSiteTemplate`
And thanks for the answer, it helped me too.

**Reply by kasperbolarsen** [reply]
Correct,  Add-PnPFileToProvisioningTemplate has be renamed in PnP.PowerShell
